cosmwasm/cli-utils.js

```diff
@@ -78,6 +82,14 @@ const addAmplifierOptions = (program, options) => {
             ),
         );
     }
+
+    program.hook('preAction', (command) => {
+        const opts = command.opts();
+
+        if (!opts.contractVersion && !opts.artifactPath) {
+            throw new Error("Either '--contractVersion' (Contract Version) or '--artifactPath' (Local Path) must be provided.");
+        }
+    });
```

Comment:
I think there is a way to add mutually exclusive options with CommanderJs: https://github.com/tj/commander.js/blob/master/examples/options-conflicts.js could you try that?